chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomsStateDifferTest.kt

```diff
@@ -0,0 +1,64 @@
+package com.pusher.chatkit.rooms.state
+
+import assertk.assertThat
+import assertk.assertions.containsExactly
+import com.pusher.chatkit.state.LeftRoom
+import com.pusher.chatkit.state.ReconnectJoinedRoom
+import com.pusher.chatkit.state.RoomUpdated
+import com.pusher.chatkit.state.State
+import io.mockk.every
+import io.mockk.mockk
+import org.reduxkotlin.GetState
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class JoinedRoomsStateDifferTest : Spek({
+
+    describe("given one room") {
+        val state = mockk<GetState<State>>(relaxed = true)
+        every { state().joinedRoomsState } returns JoinedRoomsState(
+            rooms = mapOf(roomOneId to roomOne),
+            unreadCounts = mapOf(roomOneId to 1))
+        val differ = JoinedRoomsStateDiffer(state)
+
+        describe("when room is joined") {
+            val actions = differ.toActions(
+                newRooms = listOf(roomOne, roomTwo),
+                newUnreadCounts = mapOf(roomOneId to 1, roomTwoId to 2))
+
+            it("then the result contains JoinedRoom action") {
+                assertThat(actions).containsExactly(ReconnectJoinedRoom(room = roomTwo, unreadCount = 2))
```

Comment:
```suggestion
                assertThat(actions).containsExactly(ReconnectJoinedRoom(roomTwo, unreadCount = 2))
```
minor: which such legible names named params just add noise.